### ISSUE TITLE
Redlionfish on Mac M1 (arm64)

### ISSUE THREAD

**Issue by haesleinhuepf** [rosalindfranklininstitute/RedLionfish#13]
Hi Luis @perdigao1 , hi all,

I just got a report from a collaborator that our napari-plugin distribution [devbio-napari](https://github.com/haesleinhuepf/devbio-napari) cannot be installed on Mac M1 because of an issue with RedLionFish. devbio-napari depends on redlionfish. The reported error was (I would provide more details if I could):

```
Encountered problems while solving:
  - nothing provides redlionfish needed by devbio-napari-0.1.5-pyhd8ed1ab_0
```

I know by chance that pyopencl works on Mac M1 and thus, was wondering why conda can't find a redlionfish that's compatible with this system. Can you give further insights?

No hurry btw. I removed redlionfish from the distribution temporarily so that the installation becomes functional again. I would be very happy to put it back in though!

Thanks!

Best,
Robert

**Comment by perdigao1** [comment]
Hi Robert,

Thank you for reporting that problem. I will have a look into it. I am currently fixing other issues with the code.

I have updated recently the RedLionfish in PyPI (PyOpenCL can now be installed using pip with the precompiled libraries) but not in conda  but there should be a working version there. I will check github actions if any error occurred in Mac env

Cheers
Luis

**Comment by haesleinhuepf** [comment]
> You do say that it is arm64 (not intel), not sure if conda-forge compiles for that

That might be the problem then. Maybe @jaimergp can assist us in making redlionfish deploy on Mac arm64? 🙃

**Comment by jaimergp** [comment]
The migration PR is there https://github.com/conda-forge/redlionfish-feedstock/pull/5

It needs to be reviewed and merged by @perdigao1. I have ordered the bot to re-run it to make sure it's current. Once it lands again, you can review it!

There's also a pending version update (to be merged only _after_ the osx-arm64 support PR).

**Comment by haesleinhuepf** [comment]
Amazing, thanks for the lightspeed response @jaimergp ! 🤩

**Comment by perdigao1** [comment]
OK. I updated the package to version v0.8 and went through a maze of pushes, pulls and merges. I think I managed to get it right now., and I hope it is working. Thank you @jaimergp for fixing and to @haesleinhuepf for highlighting these issues.

**Comment by jaimergp** [comment]
Issued an automated fix for some permissions issue with the macOS scripts. https://github.com/conda-forge/redlionfish-feedstock/pull/10

**Comment by jaimergp** [comment]
Btw, is `RedLionFish` using any compiled extensions? I think it qualifies for `noarch`, given that it seems to be pure Python. That would reduce the amount of builds to one or two. I do see you have conditional requirements (`ocl-icd-system` on Linux only), but that can be worked around with `noarch_platforms` as we did for `devbio-napari`. I can set this up for you if you are interested!

**Comment by perdigao1** [comment]
Yes. RedLionFish is python only. I don't know how to setup. Please feel free to setup.
